packages/server/modules/core/graph/resolvers/appTokens.ts

```diff
@@ -0,0 +1,35 @@
+import { Resolvers } from '@/modules/core/graph/generated/graphql'
+import { canCreateAppToken } from '@/modules/core/helpers/token'
+import { getTokenAppInfo } from '@/modules/core/repositories/tokens'
+import { createAppToken } from '@/modules/core/services/tokens'
+
+export = {
+  Query: {
+    async authenticatedAsApp(_parent, _args, ctx) {
+      const { appId, token } = ctx
+      if (!appId || !token) return null
+
+      return (await getTokenAppInfo({ appId, token })) || null
+    }
+  },
+  Mutation: {
+    async appTokenCreate(_parent, args, ctx) {
+      const appId = ctx.appId || ''
+      canCreateAppToken({
+        strict: true,
+        userScopes: ctx.scopes || [],
+        tokenScopes: args.token.scopes,
+        userAppId: appId,
+        tokenAppId: appId
```

Comment:
Isn't the point of the check is to ensure the user and the token app id to match?
If we're providing the same id it will surely match.